src/swap/handle_swap_sign_transaction.c

```diff
@@ -119,7 +121,75 @@ bool swap_copy_transaction_parameters(create_transaction_parameters_t* params) {
     return true;
 }
 
-bool swap_check_validity() {
+static address_t* swap_get_tx_recipient_address(void) {
+    address_t* recipient = NULL;
+
+    /*
+     *  XXX:
+     *   In case of Jetton transfer, transaction `to` address is the sender jetton wallet address.
+     *   Recipient address is the second hints in message hints.
+     */
+    if (G_context.tx_info.transaction.hints_type == TRANSACTION_TRANSFER_JETTON) {
+        if (G_context.tx_info.transaction.hints.hints_count >= 2) {
+            PRINTF("Tx recipient is a jetton wallet address\n");
+            recipient = &G_context.tx_info.transaction.hints.hints[1].address.address;
```

Comment:
Is hint length spec ordered? Is there a risk of adding a silent comment at hint[1] to fool the check ?

Reply:
Yes parsing is ordered, see [here](https://github.com/LedgerHQ/app-ton/blob/6c5ccc216226f936ad799e81102cb843966e8300/src/transaction/transaction_hints.c#L343)